crates/core/src/factories/stub_transport.rs

```diff
@@ -0,0 +1,341 @@
+    task_list: Arc<Mutex<tokio::task::JoinSet<()>>>,
+    handler: Arc<TxImpHnd>,
+    this_url: Url,
+    cmd_send: CmdSend,
+    mut cmd_recv: CmdRecv,
+) {
+    let mut con_pool = HashMap::new();
+
+    while let Some(cmd) = cmd_recv.recv().await {
+        match cmd {
+            Cmd::RegCon(url, data_send, mut data_recv) => {
+                match handler.peer_connect(url.clone()) {
+                    Err(_) => continue,
+                    Ok(preflight) => {
+                        let (s, _) = tokio::sync::oneshot::channel();
+                        let _ = data_send.send((preflight, s));
+                    }
+                }
+
+                let cmd_send2 = cmd_send.clone();
+                let url2 = url.clone();
+                task_list.lock().unwrap().spawn(async move {
+                    while let Some((data, res)) = data_recv.recv().await {
+                        if cmd_send2
+                            .send(Cmd::InData(url2.clone(), data, res))
+                            .is_err()
+                        {
+                            break;
+                        }
+                    }
+                });
+
+                con_pool.insert(
+                    url.clone(),
+                    DropSend::new(data_send, handler.clone(), url),
+                );
+            }
+            Cmd::InData(url, data, res) => {
+                if let Err(err) = handler.recv_data(url.clone(), data) {
+                    if let Some(mut data_send) = con_pool.remove(&url) {
+                        data_send.reason = Some(format!("{err:?}"));
+                    }
+                    let _ = res.send(Err(err));
+                } else {
+                    let _ = res.send(Ok(()));
+                }
+            }
+            Cmd::Disconnect(url, payload, res) => {
+                if let Some(mut data_send) = con_pool.remove(&url) {
+                    if let Some((reason, payload)) = payload {
+                        data_send.reason = Some(reason);
+                        let _ = data_send.send.send((payload, res));
+                    }
+                }
+            }
+            Cmd::Send(url, data, res) => {
+                if let Some(send) = get_stat().connect(
+                    &cmd_send,
+                    &mut con_pool,
+                    &url,
+                    &this_url,
+                ) {
+                    let _ = send.send((data, res));
+                }
+            }
+        }
+    }
+}
+
+struct Listener {
+    id: u64,
+    url: Url,
+    recv: ConRecv,
+}
+
+impl std::fmt::Debug for Listener {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Listener").field("url", &self.url).finish()
+    }
+}
+
+impl Drop for Listener {
+    fn drop(&mut self) {
+        get_stat().remove(self.id);
+    }
+}
+
+impl Listener {
+    pub fn url(&self) -> Url {
+        self.url.clone()
+    }
+}
+
+struct Stat {
```

Comment:
Quick doc comment explaining the structs in here?

Reply:
Done: https://github.com/holochain/kitsune2/pull/45/commits/117fa4d9c58a6fcc2bf095c26c8c706c69e031c4